enterprise/server/raft/replica/replica.go

```diff
@@ -232,6 +249,17 @@ func (sm *Replica) rangeCheckedSet(wb *pebble.Batch, key, val []byte) error {
 	if !keys.IsLocalKey(key) {
 		if sm.mappedRange != nil && sm.mappedRange.Contains(key) {
 			sm.rangeMu.RUnlock()
+
+			if isFileRecordKey(key) {
+				fileMetadata := &rfpb.FileMetadata{}
+				if err := proto.Unmarshal(val, fileMetadata); err != nil {
+					return err
+				}
```

Comment:
unmarshalling every key written to the replica in order to count it feels pretty expensive. I see why this is convenient though since it lets us count replayed as well as directly written files.

Would it be possible to extract this information from the key instead and skip the unmarshalling?

Reply:
Is the unmarshaling cost here worth worrying about it? 
I haven't measured it, but I imagine that cost is relatively low compared to the cost of the enclosing Raft write.

Reply:
Updated to look at the key instead. I remembered that we need to look at the key for estimating DB size anyway so there's no point trying to avoid parsing parts of the key.